### ISSUE TITLE
IEnumerable<Option<T>>.Flatten() to remove Nones from collection of Option<T> (and unwrap Somes)

### ISSUE THREAD

**Issue by voxoid0** [nlkl/Optional#30]
btw thanks for the awesome library!

Borrowing from scala Option functionality, it would be very convenient/intuitive to have a Flatten() method for a collection of Option<T> that unwraps the values, removing the Nones, as in this extension method (another name is fine with me too...Unwrap?):

```
        public static IEnumerable<T> Flatten<T>(this IEnumerable<Option<T>> options)
        {
            return options.SelectMany(o => o.ToEnumerable());
        }

```
And test:

 ```
       [Test]
        public void Flatten_Always_RemovesNonesAndUnwrapsSomes()
        {
            var options = new List<Option<int>>()
            {
                1.Some(),
                Option.None<int>(),
                3.Some(),
                Option.None<int>(),
                Option.None<int>(),
                6.Some()
            };

            var flattened = options.Flatten();

            CollectionAssert.AreEqual(new int[] { 1, 3, 6}, flattened);
        }
```

**Comment by nlkl** [comment]
Hi, and thanks for you comment and suggestion!

I am planning on merging some of the Optional.Collections functionality into Optional itself, as the collection methods are actually often very useful, but only few seem to enjoy including an extra Nuget package into their projects.

Once I get around to this, I think it is a fine idea also adding a Flatten-extension method to the collection handling functionality, as the `options.SelectMany(value => value.ToEnumerable())` pattern appears very often in actual code bases.

I will schedule this for the upcoming 4.0.0 release.

Best,
Nils

**Comment by nlkl** [comment]
Implemented by 56251c89db989d4802e9163557e4c45806bd75fc
